assets/course-theme/sidebar.js

```diff
@@ -0,0 +1,249 @@
+/**
+ * External dependencies
+ */
+import debounce from 'lodash/debounce';
+
+/**
+ * The last scroll top value.
+ *
+ * @member {number}
+ */
+let lastScrollTop = 0;
+
+/**
+ * Calculates the scroll delta.
+ */
+const getScrollDelta = () => {
+	const { scrollTop } = document.documentElement;
+	const delta = scrollTop - lastScrollTop;
+	lastScrollTop = Math.max( 0, scrollTop );
+	return delta;
+};
+
+/**
+ * Tells if the sidebar is supposed to be sticky.
+ *
+ * @return {boolean} True if it is sticky. False otherwise.
+ */
+const isStickySidebar = () =>
+	[ 'modern', 'video-full' ].some( ( templateName ) =>
+		document.body.classList.contains( `learning-mode--${ templateName }` )
+	);
```

Comment:
Could we use something like an `.is-sticky` class on the sidebar block instead? Coding it to template names makes it hard to figure out where this behavior is coming from and why it's only happening in some cases. 
With an `.is-sticky` class, we can also build it into a setting toggle in the editor later.

(This does mean changing the templates, which won't affect folks who customized it already, but that's probably for the best.)

Reply:
You're probably right. It's better to have that info close to the sidebar blocks.

Although the block is not strictly a sidebar block. It's variation of the `sensei-lms/ui` block. I'm sure there is a way to add some sort of block setting depending on the block variation. I didn't spend time to figure that out though. Didn't want to turn this into bigger project.

What I did instead is I added a custom class name in the **Advanced** section of the **Sidebar Menu** block. So it's on by default and can be removed by the user if they want to. Maybe we can turn it into a block setting with a better UI. For now I think this approach is sufficient.

Checkout the change in the sensei pro repo here: https://github.com/Automattic/sensei-pro/pull/1988

b5c8462a623219dc9414845a4674b90a481e417f